polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -465,11 +475,50 @@ impl CandidateCommitments {
 			{
 				UMPSignal::SelectCore(core_index_selector, cq_offset) =>
 					Ok(Some((core_index_selector, cq_offset))),
+				_ => Err(CommittedCandidateReceiptError::UmpSignalDecode),
 			}
 		} else {
 			Ok(None)
 		}
 	}
+
+	/// Returns the ump signals of this candidate, if any, or an error if they violate the expected
+	/// format.
+	pub fn ump_signals(
```

Comment:
This should rather be an internal method. The public methods should be  `core_selector()` and `approved_peer()` that internally call ump_signals().

Or if we are afraid it's not efficient to do that, let's create a CandidateUMPSignals struct that this function returns.  Then we can have the above methods live in there.

Reply:
good idea, added a CandidateUMPSignals